test/suites/homeshick_dir.bats

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bats
+
+load ../helper
+
+@test 'bash override homeshick_dir' {
+	castle 'dotfiles'
+	local result=$( HOMESHICK_DIR=$_TMPDIR/nowhere $HOMESHICK_FN 2>&1 >/dev/null )
+	echo "result=$result"
+	[[ "$result" =~ "/nowhere/" ]]
+}
+
+@test 'fish override homeshick_dir' {
+	[ $(type -t fish) = "file" ] || skip "fish not installed"
+	pwd
+	ls $TESTDIR
+	cmd="source ${HOMESHICK_FN_SRC%.sh}.fish; set HOMESHICK_DIR \"$_TMPDIR/nowhere\"; $HOMESHICK_FN"
```

Comment:
All three tests work by assigning an invalid path containing the string "nowhere" to `HOMESHICK_DIR`. When `$HOMESHICK_FN` is called, it will detect `HOMESHICK_DIR` is set and apply it to run a non-existent command. This causes the shell to emit an error message like this (for bash):
```
/tmp/tmp.r7X0nLSYab/home/.homesick/repos/homeshick/homeshick.sh: line 10: /tmp/tmp.r7X0nLSYab/nowhere/bin/homeshick: No such file or directory```
Hence, by checking for error output containing the path segment `/nowhere/` we know such an error message was written, and that `HOMESHICK_DIR` was applied.